### PR TITLE
Custom UI Elements Refactoring

### PR DESCRIPTION
**Pull request by pondorasti**
Classes that have been refactored:
* UITextFieldContainer
* UIDescriptorView
* UIIconView
* UIStickyView
* KifuTabBarController
* UIAnimatedButton

### CODE REVIEW

**Comment by pondorasti** [comment]
[Custom UI Refactoring Stage 1](https://trello.com/c/BrNO2xbS/119-custom-ui-refactoring-stage-1)

File: KifuSF/Protocols.swift [hunk 1]
```diff
@@ -9,6 +9,9 @@
 import Foundation
 
 protocol UIConfigurable {
+    /**
+     Poof
+     */
```
Comment (LinnierGames) [inline KifuSF/Protocols.swift:14]: Wha this??

File: KifuSF/UIAnimatedButton.swift [hunk 2]
```diff
@@ -137,7 +123,7 @@ class KFButton: UIButton, UIConfigurable {
         case .pressed:
             buttonAnimator = nil
         default:
-            let animator = UIViewPropertyAnimator(duration: KFButton.animationDuration, curve: .linear, animations: { [unowned self] in
+            let animator = UIViewPropertyAnimator(duration: UIAnimatedButton.animationDuration, curve: .linear, animations: { [unowned self] in
```
Comment (LinnierGames) [inline KifuSF/UIAnimatedButton.swift:126]: @Pondorasti be careful when you use `unowned` in your capture lists. In an animation block, `self` could be removed from memory before the animation ends

File: KifuSF/UIStickyView.swift [hunk 3]
```diff
@@ -48,38 +49,39 @@ class KFVSticky<T: UIView>: UIView, UIConfigurable {
...
-                                relation: (stickySide?.opositeSide() == .bottom) ? .greaterThanOrEqual : .equal)
-        ]
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
     }
     
-    
+    //MARK: I have no idea who wrote this code 🤷‍♂️... definitly not me 🤥
+    //FIXED: removed the ugly function, no need for somebody else to see it
```
Comment (LinnierGames) [inline KifuSF/UIStickyView.swift:58]: Is this comment current? or outdated

**Review by LinnierGames — APPROVED** [review]
Looks good

File: KifuSF/Protocols.swift (hunk 1 above)
Comment (pondorasti) [reply KifuSF/Protocols.swift:14]: Fixed!

File: KifuSF/UIStickyView.swift (hunk 3 above)
Comment (pondorasti) [reply KifuSF/UIStickyView.swift:58]: Outdated, let's just leave it there, it's part of our history :]

**Pull request merged**